.github/workflows/bench_loop.yml

```diff
@@ -0,0 +1,98 @@
+name: Benchmark loop
+
+on:
+  push:
+    branches: [ main ]
+    # Only run benchmarks on changes to following paths:
+    paths:
+      - 'risc0/**'
+      - '.github/workflows/bench_*'
+
+  # uncomment for only for testing changes to this workflow while in a PR
+  # pull_request:
+  #   branches: [ main ]
+
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  bench:
+    runs-on: [self-hosted, prod, bench, "${{ matrix.os }}", "${{ matrix.device }}"]
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [macOS]
+        feature: [default]
+        device: [cpu]
+        include:
+          - os: Linux
+            feature: cuda
+            device: nvidia_rtx_a5000
+          - os: macOS
+            feature: metal
+            device: apple_m2_pro
+
+    env:
+      CUDA_LIBRARY_PATH: /usr/lib/cuda
+      FEATURE: ${{ matrix.feature }}
+      RUSTC_WRAPPER: sccache
+
+    steps:
+      # This is a workaround from: https://github.com/actions/checkout/issues/590#issuecomment-970586842
+      - run: "git checkout -f $(git -c user.name=x -c user.email=x@x commit-tree $(git hash-object -t tree /dev/null) < /dev/null) || :"
+      - uses: actions/checkout@v3
+      - uses: ./.github/actions/rustup
+      - uses: ./.github/actions/sccache
+        with:
+          key: ${{ matrix.os }}-${{ matrix.feature }}
+      - run: cargo run --bin cargo-risczero -- risczero install
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - run: cargo run --release -F $FEATURE --example loop -- --json | tee ${{ matrix.os }}-${{ matrix.device }}.json
+        working-directory: risc0/zkvm
+      - name: Upload benchmark result
+        uses: actions/upload-artifact@v2
+        with:
+          name: benchmark-result-${{ matrix.device }}
+          path: risc0/zkvm/${{ matrix.os }}-${{ matrix.device }}.json
+
+  push_to_ghpages:
+    needs: bench
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout gh-pages repository
+        uses: actions/checkout@v3
+        with:
+          repository: risc0/ghpages
+          ref: main
+          token: ${{ secrets.BENCHMARK_TOKEN }}
+      - name: Download benchmark cpu result
+        uses: actions/download-artifact@v2
+        with:
+          name: benchmark-result-cpu
+          path: datasheet
+      - name: Download benchmark apple_m2_pro result
+        uses: actions/download-artifact@v2
+        with:
+          name: benchmark-result-apple_m2_pro
+          path: datasheet
+      - name: Download benchmark nvidia_rtx_a5000 result
+        uses: actions/download-artifact@v2
+        with:
+          name: benchmark-result-nvidia_rtx_a5000
+          path: datasheet
```

Comment:
We should be able to combine all of these into a single action, the download action is designed to download multiple assets.